### ISSUE TITLE
Without `strict` mode, reading `__v` from the outside of the class, pyright doesn't give any error while Python interpreter gives error

### ISSUE THREAD

**Issue by hyperkai** [microsoft/pyright#11194]
*Memo:
- pyright test.py (with or without `strict` mode)
- pyright 1.1.407
- python test.py
- Python 3.14.0
- Windows 11

Without `strict` mode, reading `__v` from the outside of the class, pyright doesn't give any error as shown below:

*Memo:
- A double leading underscore(__abc) can make attributes private in a class.

```python
class Cls:
    __v = 100

cls: Cls = Cls()

print(cls.__v) # No error
```

I know that with `strict` mode, pyright gives the error as shown below:

> error: "__v" is private and used outside of the class in which it is declared

But, because Python interpreter gives the error as shown below so pyright should also give error whether `strict` mode or not:

> AttributeError: 'Cls' object has no attribute '__v'

**Comment by erictraut** [comment]
This check is covered by the `reportPrivateUsage` check. You can enable this check in your pyright config without enabling all strict-mode checks.

This isn't really a type issue. Not surprisingly, the Python typing spec doesn't mandate that a type checker check for this condition.
